boa/src/profiler.rs

```diff
@@ -0,0 +1,93 @@
+#![allow(missing_copy_implementations, missing_debug_implementations)]
+
+#[cfg(feature = "profiler")]
+use measureme::{EventId, Profiler, TimingGuard};
+#[cfg(feature = "profiler")]
+use once_cell::sync::OnceCell;
+use std::fmt::{self, Debug};
+#[cfg(feature = "profiler")]
+use std::{
+    path::Path,
+    thread::{current, ThreadId},
+};
+
+/// MmapSerializatioSink is faster on macOS and Linux
+/// but FileSerializationSink is faster on Windows
+#[cfg(not(windows))]
+#[cfg(feature = "profiler")]
+type SerializationSink = measureme::MmapSerializationSink;
+#[cfg(windows)]
+#[cfg(feature = "profiler")]
+type SerializationSink = measureme::FileSerializationSink;
+#[cfg(feature = "profiler")]
+pub struct BoaProfiler {
+    profiler: Profiler<SerializationSink>,
+}
+
+#[cfg(feature = "profiler")]
+static mut INSTANCE: OnceCell<BoaProfiler> = OnceCell::new();
+
+#[cfg(feature = "profiler")]
+impl BoaProfiler {
+    pub fn start_event(&self, label: &str, category: &str) -> TimingGuard<'_, SerializationSink> {
+        let kind = self.profiler.alloc_string(category);
+        let id = EventId::from_label(self.profiler.alloc_string(label));
+        let thread_id = Self::thread_id_to_u32(current().id());
+        self.profiler
+            .start_recording_interval_event(kind, id, thread_id)
+    }
+
+    pub fn default() -> BoaProfiler {
+        let profiler = Profiler::new(Path::new("./my_trace")).unwrap();
+        BoaProfiler { profiler }
+    }
```

Comment:
We should implement the `Default` trait, and any function returning `BoaProfiler` in `BoaProfiler` should return `Self`, in case we change its name in the future, to make it easier.